src/realm/array_unsigned.cpp

```diff
@@ -91,23 +92,25 @@ void ArrayUnsigned::update_from_parent() noexcept
 
 size_t ArrayUnsigned::lower_bound(uint64_t value) const noexcept
 {
-    if (m_width == 8) {
+    auto width = get_width_from_header(get_header());
```

Comment:
This is wrong... and not needed (also slows down a bit lower bound, because m_width can be used), I thought I had changed this. Probably I did it in a different PR that I did not merge back here. The same applies to upper_bound and any other check we do in `ArrayUnsigned`. It is something forgotten from when I tried to compress also these arrays.